packages/ckeditor5-list/src/documentlist/documentlistediting.js

```diff
@@ -84,14 +92,68 @@ export default class DocumentListEditing extends Plugin {
 
 		editor.commands.add( 'indentList', new DocumentListIndentCommand( editor, 'forward' ) );
 		editor.commands.add( 'outdentList', new DocumentListIndentCommand( editor, 'backward' ) );
+
+		editor.commands.add( 'splitListItem', new DocumentListSplitCommand( editor ) );
+
+		// Overwrite the default Enter key behavior: outdent or split the list in certain cases.
+		this.listenTo( editor.editing.view.document, 'enter', ( evt, data ) => {
+			const doc = model.document;
+			const positionParent = doc.selection.getFirstPosition().parent;
+
+			if ( doc.selection.isCollapsed && positionParent.hasAttribute( 'listItemId' ) && positionParent.isEmpty ) {
+				// * a      →      * a
+				// * []     →      []
+				if ( isFirstBlockOfListItem( positionParent ) ) {
+					editor.execute( 'outdentList' );
+
+					data.preventDefault();
+					evt.stop();
+				}
+				// * a      →      * a
+				//   []     →      * []
+				else if ( isLastBlockOfListItem( positionParent ) ) {
+					editor.execute( 'splitListItem' );
+
+					data.preventDefault();
+					evt.stop();
+				}
+			}
+		}, { context: 'li' } );
+
+		// In some cases, the integration with the enter key is done after the default handler in EnterCommand.
```

Comment:
```suggestion
		// In some cases, after the default block splitting, we want to modify the new block to become a new list item instead of an additional block in the same list item.
```